crates/holochain/src/conductor/conductor.rs

```diff
@@ -1478,29 +1486,39 @@ mod app_impls {
             defer_memproofs: bool,
             ops: AppRoleResolution,
             ignore_genesis_failure: bool,
+
+            allow_throwaway_random_agent_key: bool,
         ) -> ConductorResult<InstalledApp> {
             let dpki = self.running_services().dpki.clone();
 
             // if dpki is installed, load dpki state
-            let mut dpki = if let Some(d) = dpki.as_ref() {
+            let dpki = if let Some(d) = dpki.as_ref() {
                 let lock = d.state().await;
                 Some((d.clone(), lock))
             } else {
                 None
             };
 
+            let (_, app_index) = self
+                .update_state_prime(|mut state| {
+                    let index = state.cumulative_app_installation_count;
+                    state.cumulative_app_installation_count += 1;
+                    Ok((state, index))
+                })
+                .await?;
```

Comment:
Is it correct to be incrementing this here before checking if an agent key was passed in below?

Reply:
You're right, it's better to increment only when we derive a key. And really this should just be named `agent_keys_derived`